File: exampleSources/objc/app2.m
// Objective-C — NSDictionary + blocks for transformation.

#import <Foundation/Foundation.h>

int main(int argc, const char *argv[]) {
    @autoreleasepool {
        NSDictionary *counts = @{
            @"rust":   @42,
            @"scala":  @27,
            @"python": @35,
            @"go":     @18,
        };

        NSArray *langs = [counts keysSortedByValueUsingComparator:^NSComparisonResult(NSNumber *a, NSNumber *b) {
            return [b compare:a];
        }];

        NSUInteger total = 0;
        for (NSNumber *n in [counts allValues]) {
            total += [n unsignedIntegerValue];
        }

        NSLog(@"total = %lu", (unsigned long)total);
        for (NSString *lang in langs) {
            NSLog(@"%@ -> %@", lang, counts[lang]);
        }
    }
    return 0;
}

File: exampleSources/scala/app2.scala
// Scala — collection ops on a Map + fold.

object Main extends App {
  val wordCounts = Map(
    "rust"   -> 42,
    "scala"  -> 27,
    "python" -> 35,
    "go"     -> 18,
  )

  val total = wordCounts.values.foldLeft(0)(_ + _)
  val top = wordCounts.toList
    .sortBy { case (_, c) => -c }
    .take(3)

  println(s"total = $total")
  top.foreach { case (lang, c) => println(s"$lang -> $c") }
}